src/main/java/cceclipseplugin/core/EclipseRequestManager.java

```diff
@@ -48,34 +53,41 @@ public EclipseRequestManager(DataManager dataManager, WSManager wsManager,
 	
 	@Override
 	public void finishSubscribeToProject(long id, File[] files) {
+		MetadataManager metaMgr = PluginManager.getInstance().getMetadataManager();
 		IWorkspaceRoot root = ResourcesPlugin.getWorkspace().getRoot();
 		Project p = PluginManager.getInstance().getDataManager().getSessionStorage().getProjectById(id);
 		IProject eclipseProject = root.getProject(p.getName());
-		ProjectMetadata meta =  new ProjectMetadata();
-		meta.setName(p.getName());
-		meta.setProjectID(id);
-		PluginManager.getInstance().getMetadataManager().putProjectMetadata(eclipseProject.getFullPath().toString(), meta);
+		NullProgressMonitor progressMonitor = new NullProgressMonitor();
+		
+		// create & open a new project, deleting the old one if it exists
 		try {
-			NullProgressMonitor progressMonitor = new NullProgressMonitor();
 			if (eclipseProject.exists()) {
 				eclipseProject.delete(true, true, progressMonitor);
 			}
 			eclipseProject.create(progressMonitor);
 			eclipseProject.open(progressMonitor);
-			for (File f : files) {
-				pullFileAndCreate(eclipseProject, p, f, progressMonitor);
-			}
 		} catch (CoreException e) {
 			e.printStackTrace();
 		}
+		
+		// create project metadata and write the metadata to disk
+		ProjectMetadata pmeta = new ProjectMetadata();
+		pmeta.setName(p.getName());
+		pmeta.setProjectID(id);
+		List<FileMetadata> fileMetadatas = new ArrayList<>();
+		for (File f : files) {
+			pullFileAndCreate(eclipseProject, p, f, progressMonitor);
+			fileMetadatas.add(new FileMetadata(f));
```

Comment:
For Directory Watching to be functional, the file metadata needs to be added before actually creating the file; this is an assumption I went with when writing how Directory Watching works.